AlgorithmFactory/Loader.cs

```diff
@@ -241,12 +241,6 @@ private bool TryCreateILAlgorithm(string assemblyPath, out IAlgorithm algorithmI
                     var assemblyBytes = File.ReadAllBytes(assemblyPath);
                     assembly = Assembly.Load(assemblyBytes, debugInformationBytes);
                 }
-                if (assembly == null)
```

Comment:
Gotta love R#! `Assembly.Load` and `Assembly.LoadFrom` guarantee a non-null result or an exception 👍